cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/TemporalFunctionsTest.scala

```diff
@@ -0,0 +1,1317 @@
+          considerations("If `instant~2~` occurs earlier than `instant~1~`, the resulting _Duration_ will be negative.")
+          query(
+            """UNWIND [duration.inSeconds(date("1984-10-11"), date("2015-06-24")),
+              |   duration.inSeconds(localtime("14:30"), localtime("16:30")),
+              |   duration.inSeconds(time("14:30"), date("2015-06-24")),
+              |   duration.inSeconds(datetime("2014-07-21T21:40:36.143+0200"), datetime("2015-07-21T21:40:32.142+0100")),
+              |   duration.inSeconds(datetime("2015-07-21T21:40:32.142+0100"), datetime("2014-07-21T21:40:36.143+0200"))] AS d
+              |RETURN d""".stripMargin, ResultAssertions((r) => {
+              //CYPHER_TODO
+            })) {
+            resultTable()
+          }
+        }
+      }
+    }
+    section("date(): getting the current _Date_", "functions-date-current") {
+      p(
+        """`date()` returns the current _Date_ value.
+          |If no time zone parameter is specified, the local time zone will be used.
+        """.stripMargin)
+      function("date([ {timezone} ])", "A Date.", ("A single map consisting of the following:", ""), ("timezone", "A string expression that represents the <<cypher-temporal-specify-time-zone, time zone>>"))
+      considerations("If no parameters are provided, `date()` must be invoked (`date({})` is invalid).")
+      query(
+        """RETURN date() AS currentDate""".stripMargin, ResultAssertions((r) => {
+          val now = r.columnAs[LocalDate]("currentDate").next()
+          now should be(a[LocalDate])
+        })) {
+        p("""The current date is returned.""")
+        resultTable()
+      }
+      query(
+        """RETURN date( {timezone: 'America/Los Angeles'} ) AS currentDateInLA""".stripMargin, ResultAssertions((r) => {
+          val now = r.columnAs[LocalDate]("currentDateInLA").next()
+          now should be(a[LocalDate])
+        })) {
+        p("""The current date in California is returned.""")
+        resultTable()
+      }
+      section("date.transaction()", "functions-date-current-transaction") {
+        p(
+          """`date.transaction()` returns the current _Date_ value using the `transaction` clock.
+            |This value will be the same for each invocation within the same transaction.
+            |However, a different value may be produced for different transactions.
+          """.stripMargin)
+        function("date.transaction([ timezone ])", "A Date.", ("timezone", "A string expression that represents the <<cypher-temporal-specify-time-zone, time zone>>"))
+        query(
+          """RETURN date.transaction() AS currentDate""".stripMargin, ResultAssertions((r) => {
```

Comment:
Users will see the date when we last built the documentation. That is probably absolutely OK, but I just wanted to mention it.